src/main/kotlin/de/twintorx/battleship/ui/Player.kt

```diff
@@ -59,29 +60,35 @@ class Player {
     private fun prepare() {
         val ships = Ship.getStandardShipSet()
 
-        Console.printPlaceShips(gameBoard, trackBoard, remainingEnemyHitPoints, remainingOwnHitPoints, remainingEnemyShips, remainingOwnShips)
+        if (Console.clearConsole().run { Console.input("${PlayerMessage.PLACE_OPTIONS}\n") { InputRegex.YES_OR_NO.matches(it) }.toLowerCase() == "y" }) {
 
-        while (ships.isNotEmpty()) {
-            Console.printChooseShip()
-            val option = Console.input(ships.map {
-                val name = it.value[0].type
-                "[${it.key}] ${it.value.size}x$name${" " * (11 - name.length)}(Size:${it.value[0].size})\n"
-            }.joinToString("")) {
-                InputRegex.SELECT_SHIP.matches(it) && ships.containsKey(it.toInt())
-            }.toInt()
+            Console.printPlaceShips(gameBoard, trackBoard, remainingEnemyHitPoints, remainingOwnHitPoints, remainingEnemyShips, remainingOwnShips)
 
-            Console.eraseLastLines(ships.size + 2)
+            while (ships.isNotEmpty()) {
+                Console.printChooseShip()
+                val option = Console.input(ships.map {
+                    val name = it.value[0].type
+                    "[${it.key}] ${it.value.size}x$name${" " * (11 - name.length)}(Size:${it.value[0].size})\n"
+                }.joinToString("")) {
+                    InputRegex.SELECT_SHIP.matches(it) && ships.containsKey(it.toInt())
+                }.toInt()
 
-            with(ships[option]!!) {
-                placeShip(this[0]).run {
-                    Console.printPlaceShips(gameBoard, trackBoard, remainingEnemyHitPoints, remainingOwnHitPoints, remainingEnemyShips, remainingOwnShips)
-                }
-                removeAt(0)
+                Console.eraseLastLines(ships.size + 2)
+
+                with(ships[option]!!) {
```

Comment:
ships[option]?.let{
...
}

das führt den let-block nur aus wenn des not null ist